core-bundle/src/Routing/Page/PageRoute.php

```diff
@@ -47,17 +47,21 @@ public function __construct(PageModel $pageModel, string $path = '', array $defa
     {
         $pageModel->loadDetails();
 
-        $defaults = array_merge(
-            [
-                '_token_check' => true,
-                '_controller' => 'Contao\FrontendIndex::renderPage',
-                '_scope' => ContaoCoreBundle::SCOPE_FRONTEND,
-                '_locale' => LocaleUtil::formatAsLanguageTag($pageModel->rootLanguage),
-                '_format' => 'html',
-            ],
-            $defaults
-        );
+        $initialDefaults = [
+            '_token_check' => true,
+            '_controller' => 'Contao\FrontendIndex::renderPage',
+            '_scope' => ContaoCoreBundle::SCOPE_FRONTEND,
+            '_format' => 'html',
+            '_canonical_route' => 'tl_page.'.$pageModel->id,
+        ];
+
+        if ($pageModel->rootLanguage) {
```

Comment:
why do we need an if-statement here? The page model always has a `rootLanguage`

Reply:
Maybe the unit test does not set it everywhere?

Reply:
most likely, yes. But we're always calling `loadDetails()` on PageModel so it must be there. I'll do a PR to fix this.